my_enumerable_module.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+# Extending Enumerable module with replicated methods for the following:
+# each, each_with_index, select, all?, none?, any?, count, map and inject
+#
+module Enumerable
+  def my_each
+    arr = self # this is to prevent a redundant-use-of-self error
+    0.upto(arr.length - 1) do |i|
+      yield(arr[i])
+    end
+  end
+
+  def my_each_with_index
+    arr = self # this is to prevent a redundant-use-of-self error
+    0.upto(arr.length - 1) do |i|
+      yield(arr[i], i)
+    end
+  end
+
+  def my_select
+    arr = self # this is to prevent a redundant-use-of-self error
+    result = []
+    0.upto(arr.length - 1) do |i|
+      result << arr[i] if yield(arr[i])
+    end
+    result
+  end
+
+  def my_all?
+    arr = self # this is to prevent a redundant-use-of-self error
+    response = true
+    0.upto(arr.length - 1) do |i|
+      response = false unless yield(arr[i])
+    end
+    response
+  end
+
+  def my_any?
+    arr = self # this is to prevent a redundant-use-of-self error
+    response = false
+    0.upto(arr.length - 1) do |i|
+      response = true if yield(arr[i])
+    end
+    response
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern